lib/screens/practice/recognition_screen.dart

```diff
@@ -19,14 +19,38 @@ class _RecognitionScreenState extends State<RecognitionScreen> {
 
 3. Tap the hand sign that matches the number word!""";
 
+  int randomNumber = 0;
+  late List<int> choices;
+
   @override
   void initState() {
     super.initState();
+    _generateRandomNumberAndChoices();
     WidgetsBinding.instance.addPostFrameCallback((_) {
       _showInstructions();
     });
   }
 
+  void _generateRandomNumberAndChoices() {
+    setState(() {
+      randomNumber = Random().nextInt(10); // Generate random number (0-9)
+
+      // Generate 3 unique incorrect numbers
+      final random = Random();
+      final incorrectNumbers = <int>{};
+      while (incorrectNumbers.length < 3) {
+        int randomChoice = random.nextInt(10);
+        if (randomChoice != randomNumber) {
+          incorrectNumbers.add(randomChoice);
+        }
+      }

```

Comment:
Is there other way? (Performance concern) since while loop could run many times before finding 3 valid numbers.

Reply:
I think this is fine since 3 iterations wont take much memory to do

Reply:
It's about the potential long-running loops. This is fine for now, but how about we go from 0-9 to 0-99. 

GEFN 🐐

Reply:
When we go from 0-9 to 0-99 we will only loop for 3 times still because the number of incorrect choices to be generated are still 3 not 99.